From 602dcd12bfb34f6be0fc2c7aa45ac838c22869e1 Mon Sep 17 00:00:00 2001
From: Adrien Ricciardi <aricciardi@baylibre.com>
Date: Mon, 18 Sep 2023 10:44:34 +0200
Subject: [PATCH 66/70] examples: renesas: memory_unlocker: Added the flashable
 firmware generation steps.

The 'build.sh' script now also generates the '_build/App_CDD_ICCOM_S4_Sample_G4MH.srec'
firmware to provide to the Flash_Tools 'burn.sh' script.

Signed-off-by: Adrien Ricciardi <aricciardi@baylibre.com>
---
 .../renesas/memory_unlocker/G4MH_Head.srec    |  7 ++++++
 examples/renesas/memory_unlocker/build.sh     | 23 ++++++++++++-------
 2 files changed, 22 insertions(+), 8 deletions(-)
 create mode 100644 examples/renesas/memory_unlocker/G4MH_Head.srec

diff --git a/examples/renesas/memory_unlocker/G4MH_Head.srec b/examples/renesas/memory_unlocker/G4MH_Head.srec
new file mode 100644
index 00000000..8b3b240c
--- /dev/null
+++ b/examples/renesas/memory_unlocker/G4MH_Head.srec
@@ -0,0 +1,7 @@
+S00F000064756D6D795F67346D680000F5
+S3150000000000040000000400001000813D00007000A4
+S31500000010FFFF7F70000000000F0C0C0C000F0000AB
+S31500000020FF0F0000000000020003000300000000B4
+S31500000200FFFFFF0F0000500000005000EFFFFFFF50
+S3110000021000006000FFFFFFFFFFFFFFFF84
+S70500000000FA
diff --git a/examples/renesas/memory_unlocker/build.sh b/examples/renesas/memory_unlocker/build.sh
index 649f5e7d..805af2db 100644
--- a/examples/renesas/memory_unlocker/build.sh
+++ b/examples/renesas/memory_unlocker/build.sh
@@ -1,17 +1,24 @@
 #!/usr/bin/env bash
 
-#stop on errors
 set -e
 
-if [[ ! -d "_build" ]]
-then
-    mkdir _build
-fi
+PROGRAM_NAME=memory_unlocker
+FLASH_TOOL_PROGRAM_NAME=App_CDD_ICCOM_S4_Sample_G4MH
 
 echo "*** Run Goil ***"
-goil --target=renesas/g4mh --templates=../../../goil/templates/ memory_unlocker.oil
-cd _build
+mkdir -p _build
+goil --target=renesas/g4mh --templates=../../../goil/templates/ ${PROGRAM_NAME}.oil
+
 echo "*** Run CMake ***"
-cmake -G "Unix Makefiles" -D CMAKE_TOOLCHAIN_FILE=../memory_unlocker/compiler.cmake ..
+cd _build
+cmake -G "Unix Makefiles" -D CMAKE_TOOLCHAIN_FILE=../${PROGRAM_NAME}/compiler.cmake ..
+
 echo "*** Run Make ***"
 make
+
+# Create the S-Record file to provide to Flash_Tool to set the G4MH core default firmware on board boot
+# The flashing procedure expects 32-bit addresses, so use S3 records instead of S1
+objcopy -O srec --srec-forceS3 ${PROGRAM_NAME}_exe.abs ${FLASH_TOOL_PROGRAM_NAME}.s3
+# Add a specific header (it has been downloaded here : https://raw.githubusercontent.com/renesas-rcar/whitebox-sdk/v4.x/mcu/G4MH_Head.srec)
+rlink ../G4MH_Head.srec ${FLASH_TOOL_PROGRAM_NAME}.s3 -FOrm=SType -OUtput=${FLASH_TOOL_PROGRAM_NAME}.srec
+rm -f ${FLASH_TOOL_PROGRAM_NAME}.s3
-- 
2.34.1

